core/storage/trie/trie_storage.hpp

```diff
@@ -30,8 +33,13 @@ namespace kagome::storage::trie {
      */
     virtual outcome::result<std::unique_ptr<PersistentTrieBatch>>
     getPersistentBatchAt(const RootHash &root) = 0;
+
     virtual outcome::result<std::unique_ptr<EphemeralTrieBatch>>
     getEphemeralBatchAt(const RootHash &root) const = 0;
+
+    virtual outcome::result<std::unique_ptr<EphemeralTrieBatch>> getProofReaderBatchAt(
+        const RootHash &root,
+        const OnNodeLoaded &on_node_loaded) const = 0;
```

Comment:
Is it just "overload with/without callback" to avoid changing tests/mocks?

Reply:
Partly to avoid changing existing code, partly for abstraction -- this is supposed to be a different batch type, it's just implemented using an Ephemeral batch.